### PR TITLE
feat: Add plothist

### PR DESCRIPTION
**Pull request by matthewfeickert**
* c.f. https://github.com/cyrraz/plothist
* Recipe generated with grayskull via 'grayskull pypi plothist' and then edited to add additional information.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [N/A] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/plothist/meta.yaml
Comment (matthewfeickert) [inline recipes/plothist/meta.yaml:54]: @cyrraz @0ctagon Can you add a comment as a post on the PR if you are willing to be a maintainer on this conda-forge recipe (required before a conda-forge maintainer can review this).

Comment (0ctagon) [reply recipes/plothist/meta.yaml:54]: Thank you for the PR!
Is it possible to remove @cyrraz, as I'm the only long term maintainer currently?

File: recipes/plothist/meta.yaml
Comment (matthewfeickert) [inline recipes/plothist/meta.yaml:33]: c.f. https://github.com/cyrraz/plothist/issues/175

File: recipes/plothist/meta.yaml
Comment (0ctagon) [reply recipes/plothist/meta.yaml]: This test is for plothist developers only, it's not detrimental for the user so I don't think it's a problem to remove it. I can also change its functionality if it's needed here.

Comment (matthewfeickert) [reply recipes/plothist/meta.yaml]: @0ctagon, no that's fine. It was just automatically discovered through `grayskull`, but there's not requirement that it get provided through `conda-forge`.

File: recipes/plothist/meta.yaml
Comment (matthewfeickert) [reply recipes/plothist/meta.yaml]: Done. The conda-forge linter requires you to comment as a standalone PR comment with something like "I am willing to be listed as a maintainer."

File: recipes/plothist/meta.yaml
Comment (matthewfeickert) [inline recipes/plothist/meta.yaml:15]: Failing on Windows for `noarch` Python projects is allowed given what I've been told by conda-forge core people in the past so this is fine.

**Comment by 0ctagon** [comment]
I am willing to be listed as a maintainer.

**Comment by matthewfeickert** [comment]
@conda-forge/help-python, ready for review!

File: recipes/plothist/meta.yaml
Comment (0ctagon) [reply recipes/plothist/meta.yaml:33]: Should we wait for https://github.com/cyrraz/plothist/pull/176 to be merged and a new small release? Or is it ok to be merged now?

Comment (matthewfeickert) [reply recipes/plothist/meta.yaml:33]: Conda-forge will create new release PRs automatically on the `conda-forge/plothist-feedstock` that will be created once this is merged, so no real need here to wait now. You can just remove those dependencies in the following release PR.

**Review by 0ctagon — APPROVED** [review]
Thank you for the PR!

File: recipes/plothist/meta.yaml
Comment (ocefpaf) [reply recipes/plothist/meta.yaml:15]: In this particular case the package won't be installable on Windows. If that is by design, we can merge this. However, it does seem like this package can be OS-noarch. See https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-packages-with-os-specific-dependencies for more details.

Comment (matthewfeickert) [reply recipes/plothist/meta.yaml:15]: @ocefpaf In that case, given https://github.com/conda-forge/staged-recipes/pull/27414#discussion_r1734616076 maybe we can wait for @0ctagon to make a patch release with https://github.com/cyrraz/plothist/pull/176 merged and then we can have that patch release be the one that generates the feedstock (so we'd update the version and sha256).

Comment (ocefpaf) [reply recipes/plothist/meta.yaml:15]: B/c there is a patch already we can:

- ignore that, merge it, and you'all fix it as soon as there is a new release
- backport the patch now

I don't think it is worth making it OS-noarch b/c the fix is there already.

Comment (matthewfeickert) [reply recipes/plothist/meta.yaml:15]: > ignore that, merge it, and you'all fix it as soon as there is a new release

@ocefpaf I'm find with this suggestion. There should be a new patch release on PyPI before the end of the week so it would only be broken on Windows for a few days.

File: recipes/plothist/meta.yaml
Comment (0ctagon) [reply recipes/plothist/meta.yaml:33]: I am not familiar with the workflow of conda-forge. We created the new release v1.2.6, but not PR got automatically created in https://github.com/conda-forge/plothist-feedstock. Should I do what the readme says (fork + PR) or did we miss a setup that would automatically create the PR?

Comment (0ctagon) [reply recipes/plothist/meta.yaml:33]: Thank you, it worked perfectly https://github.com/conda-forge/plothist-feedstock/pull/1 :smile:

**Pull request merged**
